### PR TITLE
Add test coverage for Out Vars in global code in remaining executable statements.

### LINKED ISSUE #14113: Out Variable tests for Script should cover statements that currently aren't covered by GlobalCode_* unit-tests
Basically GlobalCode_\* unit-tests should cover all the different statements that are covered by Scope_\* unit-tests

### PR DESCRIPTION
**Pull request by AlekseyTs**
Closes #14113.

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 1]
```diff
@@ -12108,6 +12141,55 @@ static bool Print(object x)
...
+public class X
+{
+    public static void Main()
+    {
+        Test1();
+    }
+
+    static void Test1()
+    {
+        var res = from x1 in new[] { TakeOutParam(1, out var y1) && Print(y1) ? 1 : 0}
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: >1 [](start = 80, length = 1)

Maybe change the second `1` to `2` so they look distinct in the output.

Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Will do.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 2]
```diff
@@ -21409,6 +21491,1980 @@ static class StaticType {}
...
+    return true;
+}
+
+static bool TakeOutParam<T>(T y, out T x) 
+{
+    x = y;
+    return true;
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe.WithScriptClassName("Script"), parseOptions: TestOptions.Script);
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: I didn't understand the value of calling `WithScriptClassName`, given that the default class name is already "Script".

Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Explicitness.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Done.

**Comment by AlekseyTs** [comment]
Thanks! Merging with a single sign-off because this is a test only change.

**Pull request merged**